src/rootPages/Designer/ui_work_datacollection_workspace_properties.js

```diff
@@ -0,0 +1,1036 @@
+         $$(ids.fixSelect).define("value", fixSelect);
+         $$(ids.fixSelect).refresh();
+      }
+
+      initPopupEditors() {
+         this.FilterComponent.init({
+            // when we make a change in the popups we want to make sure we save the new workspace to the properties to do so just fire an onChange event
+            onChange: this.onFilterChange,
+         });
+
+         this.PopupSortFieldComponent.init(this.AB);
+      }
+
+      selectSource(datasourceID, oldId) {
+         const ids = this.ids;
+         const selectedDatasource = $$(ids.dataSource)
+            .getList()
+            .getItem(datasourceID);
+
+         if (selectedDatasource && selectedDatasource.disabled) {
+            // prevents re-calling onChange from itself
+            $$(ids.dataSource).blockEvent();
+            $$(ids.dataSource).setValue(oldId || "");
+            $$(ids.dataSource).unblockEvent();
+         }
+
+         // Set settings.datasourceID
+         const dcSettings = this.CurrentDatacollection.toObj() || {};
+         dcSettings.settings = dcSettings.settings || {};
+         dcSettings.settings.datasourceID = datasourceID;
+         this.CurrentDatacollection.fromValues(dcSettings);
+
+         if (!selectedDatasource.isQuery) {
+            // populate link data collection options
+            this.initLinkDatacollectionOptions();
+
+            // populate link fields
+            this.initLinkFieldOptions(
+               this.CurrentDatacollection?.datacollectionLink?.id || null
+            );
+
+            // re-create filter & sort popups
+            this.initPopupEditors();
+
+            // populate filter & sort popups
+            this.populatePopupEditors();
+
+            this.populateBadgeNumber();
+
+            // show options
+            $$(ids.filterPanel).show();
+            $$(ids.sortPanel).show();
+         } else {
+            // hide options
+            $$(ids.filterPanel).hide();
+            $$(ids.sortPanel).hide();
+         }
+      }
+
+      showFilterPopup($button) {
+         this.filter_popup.show($button, null, { pos: "top" });
+      }
+
+      showSortPopup($button) {
+         this.PopupSortFieldComponent.show($button, null, {
+            pos: "top",
+         });
+      }
+
+      onFilterChange() {
+         const datacollection = this.CurrentDatacollection;
+         if (!datacollection) return;
+
+         const filterValues = this.FilterComponent.getValue();
+
+         datacollection.settings.objectWorkspace.filterConditions =
+            filterValues || { glue: "and", rules: [] };
+
+         let allCompconste = true;
+         filterValues.rules.forEach((f) => {
+            // if all 3 fields are present, we are good.
+            if (
+               f.key &&
+               f.rule &&
+               (f.value ||
+                  // these rules do not have input value
+                  f.rule == "is_current_user" ||
```

Comment:
it looks like the `FilterComplex.isConditionComplete()` method isn't as complete as this check here.  Go ahead and include these checks into there as well.